File: Python/Aula_R_07.py
#1
try:
    resultado = 10/0
except ZeroDivisionError:
    print("Erro: Divisão por zero não é permitida")

#2    
lista = [1,2,3]
try:
  elemento = lista[5]
except IndexError:
  print("Erro: indice fora dos limites da lista")

#3
dicionario = {"a":1,"b":2,"c":3}
try:
  elemento = dicionario["d"]
except KeyError:
  print("Chave inexistente")

#EXTRA
def main():
  try:
    num1 = float(input("Digite o primeiro numero: "))
    num2 = float(input("Digite o segundo numero: "))
    
    resultado = num1 / num2
  except ZeroDivisionError as e:
    print("Erro: Divisao por zero nao permitida")
  
  except ValueError as e:
    print("Nao e possivel dividir valores em String")
  
  except Exeption as e:
    print(f"Ocorreu um erro inesperado. Tente novamente mais tarde!{e}")
    
  else:
    print(f"O resultado da divisao e: {resultado}")
    
  finally:
    print("Programa Finalizado!")

main()

#4
palavra = "abc"
try:
  palavra = int(palavra)
except ValueError:
  print("Erro: Conversao de string pata inteiro falhou!")

#5
dicionario = {"a":[1,2,3],"b":[4,5,6]}
try:
  elemrnto = dicionario["c"][0]
except KeyError:
  print("Erro: Chave inexistente no dicionario")
except IndexError:
  print("Erro: Indice fora dos limites da lista")

#6

File: R/Aula_R_07.R
#1
tryCatch({
  resultado <- 10/0
}, error = function(e){
  print("Erro: Divisão por zero não é permitida")
})

#2
lista <- c(1,2,3)
tryCatch({
  elemento <- lista[5]
}, error = function(e){
   print("Erro: indice fora dos limites da lista")
})

#3
dicionario <- list(a=1, b=2, c=3)
tryCatch({
  elemento <- dicionario[["d"]]
},error = function(e){
  print("Erro: essa chave não existe no dicionário")
})

#4
palavra <- "abc"
tryCatch({
  palavra <- as.integer(palavra)
},warning = function(w){
  print("Erro: Conversao de string pata inteiro falhou!")
})

#5
dicionario <- list(a=c(1,2,3), b=c(4,5,6))
tryCatch({
  dicionario <- dicionario[["c"]][1]
},error = function(e){
  if(inherits(e,"simpleError")){
    print("Erro: Chave inexistente no dicionario ou indice fora dos limites da lista")
  }
})

#6
